NitroxPatcher/Patches/Dynamic/AttackCyclops_UpdateAggression_Patch.cs

```diff
@@ -0,0 +1,105 @@
+                                                new CodeMatch(OpCodes.Ldnull),
+                                                new CodeMatch(OpCodes.Call),
+                                                new CodeMatch(OpCodes.Brfalse),
+                                                new CodeMatch(OpCodes.Ldsfld),
+                                                new CodeMatch(OpCodes.Callvirt, Reflect.Property((Player t) => t.currentSub).GetGetMethod())
+                                            ]).RemoveInstructions(25)
+                                            .InsertAndAdvance([
+                                                new CodeInstruction(OpCodes.Ldarg_0),
+                                                new CodeInstruction(OpCodes.Call, Reflect.Method(() => FindClosestCyclopsNoiseManagerIfAny(default)))
+                                            ]).InstructionEnumeration();
+    }
+
+    public static bool Prefix(AttackCyclops __instance)
+    {
+        return !__instance.TryGetNitroxId(out NitroxId creatureId) ||
+               Resolve<SimulationOwnership>().HasAnyLockType(creatureId);
+    }
+
+    public static void Postfix(AttackCyclops __instance)
+    {
+        if (__instance.currentTarget && __instance.TryGetNitroxId(out NitroxId creatureId) &&
+            __instance.currentTarget.TryGetNitroxId(out NitroxId targetId) &&
+            Resolve<SimulationOwnership>().HasAnyLockType(creatureId))
+        {
+            float aggressiveToNoise = __instance.aggressiveToNoise.Value;
+
+            Resolve<IPacketSender>().Send(new AttackCyclopsTargetChanged(creatureId, targetId, aggressiveToNoise));
+            ErrorMessage.AddMessage($"[SEND] {__instance.gameObject.name} attacks {__instance.currentTarget.name}");
+        }
+    }
+
+    public static CyclopsNoiseManager FindClosestCyclopsNoiseManagerIfAny(AttackCyclops attackCyclops)
+    {
+        // Cyclops are marked with EcoTargetType.Whale
+        IEcoTarget ecoTarget = EcoRegionManager.main.FindNearestTarget(EcoTargetType.Whale, attackCyclops.transform.position, IsTargetAValidInhabitedCyclops, 2);
+        if (ecoTarget == null)
+        {
+            return null;
+        }
+        return ecoTarget.GetGameObject().GetComponent<CyclopsNoiseManager>();
+    }
+
+    public static bool IsTargetAValidInhabitedCyclops(IEcoTarget target)
+    {
+        return IsTargetAValidInhabitedCyclops(target.GetGameObject());
+    }
+
+    public static bool IsTargetAValidInhabitedCyclops(GameObject targetObject)
+    {
+        // Is a Cyclops
+        if (!targetObject.GetComponent<CyclopsNoiseManager>())
+        {
+            return false;
+        }
+
+        // Has the local player inside
+        if (Player.main && Player.main.currentSub && Player.main.currentSub.gameObject == targetObject)
+        {
+            return true;
+        }
+
+        // Has a remote player inside
+        return targetObject.GetComponentInChildren<RemotePlayerIdentifier>(true);
```

Comment:
Two cases here:
- Remote player is in Cyclops' root
- Remote player is FAR in the hierarchy, parented to the driver seat

Current issue: remote player is never unparented from the driver seat even after leaving it (???) so this would need a fix in another PR I guess

Possible solutions (that would come in the same PR mentionned above:
- have a MB (either MultiplayerCyclops or a new one) store in a list all players (local and remote) which are currently in the cyclops, and use this list instead of the components search
- One loop in the first children layer, and one in the driver seat layer

Reply:
If it's only the two cases could we either check only all direct children of the cyclops and from the driver seat down?